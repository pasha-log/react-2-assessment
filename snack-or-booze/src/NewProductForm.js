import React, { useState } from 'react';
import { Form, FormGroup, Label, Input, FormText, Button, Card, CardBody, CardTitle } from 'reactstrap';
import { useHistory } from 'react-router-dom';
import './NewProductForm.css';

const NewProductForm = ({ addProduct }) => {
	let history = useHistory();
	const INITIAL_STATE = {
		productType: '',
		name: '',
		description: '',
		recipe: '',
		serve: ''
	};
	const [ formData, setFormData ] = useState(INITIAL_STATE);
	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addProduct({ ...formData });
		setFormData(INITIAL_STATE);
		history.push('/');
	};
	return (
		<section className="col-md-4">
			<Card>
				<CardBody>
					<CardTitle className="font-weight-bold text-center">New Menu Item Form</CardTitle>
					<Form onSubmit={handleSubmit}>
						<Label>Product Type</Label>
						<FormGroup check>
							<Input name="productType" type="radio" value="snacks" onChange={handleChange} />{' '}
							<Label check>Snack</Label>
						</FormGroup>
						<FormGroup check>
							<Input name="productType" type="radio" value="drinks" onChange={handleChange} />{' '}
							<Label check>Drink</Label>
						</FormGroup>
						<FormGroup>
							<Label for="name">Product name</Label>
							<FormText>What is the name of the product?</FormText>
							<Input id="name" name="name" value={formData.name} onChange={handleChange} />
						</FormGroup>
						<FormGroup>
							<Label for="description">Product description</Label>
							<FormText>What is the description of the product?</FormText>
							<Input
								id="description"
								name="description"
								type="textarea"
								value={formData.description}
								onChange={handleChange}
							/>
						</FormGroup>
						<FormGroup>
							<Label for="recipe">Product recipe</Label>
							<FormText>What is the recipe of the product?</FormText>
							<Input
								id="recipe"
								name="recipe"
								type="textarea"
								value={formData.recipe}
								onChange={handleChange}
							/>
						</FormGroup>
						<FormGroup>
							<Label for="serve">Product serve</Label>
							<FormText>How is the product served?</FormText>
							<Input
								id="serve"
								name="serve"
								type="textarea"
								value={formData.serve}
								onChange={handleChange}
							/>
						</FormGroup>
						<Button onClick={handleSubmit}>Submit</Button>
					</Form>
				</CardBody>
			</Card>
		</section>
	);
};

export default NewProductForm;
