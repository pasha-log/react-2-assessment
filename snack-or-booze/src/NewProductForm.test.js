import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import NewProductForm from './NewProductForm';

it('renders without crashing', function() {
	render(<NewProductForm />);
});

it('matches snapshot', function() {
	const { asFragment } = render(<NewProductForm />);
	expect(asFragment()).toMatchSnapshot();
});
