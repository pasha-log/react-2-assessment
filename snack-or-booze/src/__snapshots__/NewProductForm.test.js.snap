// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          New Menu Item Form
        </h5>
        <form
          class=""
        >
          <label
            class=""
          >
            Product Type
          </label>
          <div
            class="position-relative form-check"
          >
            <input
              class="form-check-input"
              name="productType"
              type="radio"
              value="snacks"
            />
             
            <label
              class="form-check-label"
            >
              Snack
            </label>
          </div>
          <div
            class="position-relative form-check"
          >
            <input
              class="form-check-input"
              name="productType"
              type="radio"
              value="drinks"
            />
             
            <label
              class="form-check-label"
            >
              Drink
            </label>
          </div>
          <div
            class="position-relative form-group"
          >
            <label
              class=""
              for="name"
            >
              Product name
            </label>
            <small
              class="form-text text-muted"
            >
              What is the name of the product?
            </small>
            <input
              class="form-control"
              id="name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="position-relative form-group"
          >
            <label
              class=""
              for="description"
            >
              Product description
            </label>
            <small
              class="form-text text-muted"
            >
              What is the description of the product?
            </small>
            <textarea
              class="form-control"
              id="description"
              name="description"
            />
          </div>
          <div
            class="position-relative form-group"
          >
            <label
              class=""
              for="recipe"
            >
              Product recipe
            </label>
            <small
              class="form-text text-muted"
            >
              What is the recipe of the product?
            </small>
            <textarea
              class="form-control"
              id="recipe"
              name="recipe"
            />
          </div>
          <div
            class="position-relative form-group"
          >
            <label
              class=""
              for="serve"
            >
              Product serve
            </label>
            <small
              class="form-text text-muted"
            >
              How is the product served?
            </small>
            <textarea
              class="form-control"
              id="serve"
              name="serve"
            />
          </div>
          <button
            class="btn btn-secondary"
            type="button"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
